regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py

```diff
@@ -0,0 +1,128 @@
+        # two num1's with same type
+        bad = [("num1", int), ("num1", int), ("num2", int)]
+        with self.assertRaisesRegexp(Exception, "cannot be more than one character"):
+            self.context.frame.import_csv(self.dataset, bad)
+
+    def test_schema_invalid_type(self):
+        """CsvFile cration with a schema of invalid type fails."""
+        bad_schema = -77
+        with self.assertRaisesRegexp(Exception, "cannot be more than one character"):
+            self.context.frame.import_csv(self.dataset, bad_schema)
+
+    def test_schema_invalid_format(self):
+        """CsvFile creation fails with a malformed schema."""
+        bad_schema = [int, int, float, float, str]
+        with self.assertRaisesRegexp(Exception, "cannot be more than one character"):
+            self.context.frame.import_csv(self.dataset, bad_schema)
+
+    def test_frame_delim_colon(self):
+        """Test building a frame with a colon delimiter."""
+        dataset_passwd = self.get_file("passwd.csv")
+        passwd_schema = [("userid", str),
+                         ("password", str),
+                         ("usrNum", int),
+                         ("grpNum", int),
+                         ("userName", str),
+                         ("home", str),
+                         ("shell", str)]
+        passwd_frame = self.context.frame.import_csv(dataset_passwd, schema=passwd_schema, delimiter=':')
+        self.assertEqual(len(passwd_frame.take(1).data[0]), len(passwd_schema))
+
+    def test_frame_delim_tab(self):
+        """Test building a frame with a tab delimiter."""
+        dataset_delimT = self.get_file("delimTest1.tsv")
+        white_schema = [("col_A", int),
+                        ("col_B", long),
+                        ("col_3", float),
+                        ("Double", float),
+                        ("Text", str)]
+        frame = self.context.frame.import_csv(dataset_delimT, schema=white_schema, delimiter='\t')
+        self.assertEqual(len(frame.take(1).data[0]), len(white_schema))
+
+    def test_delimiter_none(self):
+        """Test a delimiter of None errors."""
+        with self.assertRaisesRegexp(Exception, "delimiter"):
+            self.context.frame.import_csv(self.dataset, self.schema, delimiter=None)
+
+    def test_delimiter_empty(self):
+        """Test an empty delimiter errors."""
+        with self.assertRaisesRegexp(Exception, "delimiter"):
+            self.context.frame.import_csv(self.dataset, self.schema, delimiter="")
+
+    def test_header(self):
+        """Test the baseline number of row without modification is correct."""
+        frame_with_header = self.context.frame.import_csv(
+            self.dataset, schema=self.schema, header=True)
+        frame_without_header = self.context.frame.import_csv(self.dataset, schema=self.schema, header=False)
+        self.assertEqual(len(frame_with_header.take(sys.maxint).data), len(frame_without_header.take(sys.maxint).data) - 1) 
```

Comment:
Again, compare the data.  What if it dropped the last row and not the header?  I know, an unlikely what-if, but we should tighten the coverage when going to the effort of setting up the testcase.